### PR TITLE
throw 401 error when user doesn't write login and password

### PR DESCRIPTION
**Pull request by Fenydude**

### CODE REVIEW

**Review by OlgaEvdokimova — DISMISSED** [review]

File: src/main/java/by/exadel/internship/controller/AuthController.java [hunk 1]
```diff
@@ -29,10 +30,15 @@
...
-                .authenticate(new UsernamePasswordAuthenticationToken(
-                        loginRequest.getLogin(),
-                        loginRequest.getPassword()));
+        Authentication authentication;
+        try {
+             authentication = authenticationManager
+                    .authenticate(new UsernamePasswordAuthenticationToken(
+                            loginRequest.getLogin(),
+                            loginRequest.getPassword()));
+        }catch (Exception e){
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/controller/AuthController.java]: What exception is actually thrown here? Isn't it BadCredentialsException by default?

Comment (Fenydude) [reply src/main/java/by/exadel/internship/controller/AuthController.java]: Actually we got NotFoundException, if we can't find user by login.

Comment (SergeyKoval) [reply src/main/java/by/exadel/internship/controller/AuthController.java]: So it should some more specific exception I believe

Comment (Fenydude) [reply src/main/java/by/exadel/internship/controller/AuthController.java]: Done. That was hard xD

File: src/main/java/by/exadel/internship/exception_handing/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -42,7 +43,14 @@
 
     @ExceptionHandler
     public ResponseEntity<IncorrectData> handleException(
-            BadCredentialsException exception) {
+            AuthenticationException exception) {
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/exception_handing/GlobalExceptionHandler.java:46]: You can use one @ExceptionHandler to handle both exceptions

File: src/main/java/by/exadel/internship/config/CustomAuthentication.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCrypt;
+import org.springframework.stereotype.Component;
+
+
+@Component
+@RequiredArgsConstructor
+public class CustomAuthentication implements AuthenticationProvider {
```
Comment (SergeyKoval) [inline src/main/java/by/exadel/internship/config/CustomAuthentication.java]:
```suggestion
public class InternshipAuthentication implements AuthenticationProvider {
```

Comment (Fenydude) [reply src/main/java/by/exadel/internship/config/CustomAuthentication.java]: Done

File: src/main/java/by/exadel/internship/exception_handing/GlobalExceptionHandler.java (hunk 2 above)
Comment (Fenydude) [reply src/main/java/by/exadel/internship/exception_handing/GlobalExceptionHandler.java:46]: Done

**Review by SergeyKoval — APPROVED** [review]

**Review by ShirshovAlex — APPROVED** [review]

**Pull request merged**